### ISSUE TITLE
Migrate to annotation processors

### ISSUE THREAD

**Issue by hpehl** [ekuefler/gwt-supereventbus#13]
Since GWT code generation will most likely be nuked with GWT 3.0, are there any plans to migrate the code generation part to annotation processors?

**Comment by ekuefler** [comment]
Yeah, I've been working a lot with annotation processors lately and I don't think this will be hard to do. I'll wait until there are more details announced about GWT 3.0 before doing the work, but it'll definitely happen if it needs to.

**Comment by muksihs** [comment]
Any updates for moving to strictly annotation based/non-reflection/non gwt-create approach?

**Comment by ekuefler** [comment]
As far as I know there haven't been any concrete updates on GWT 3.0 yet. Have you heard otherwise, or is there some other reason you need to move away from GWT.create sooner rather than later?

**Comment by gkdn** [comment]
@ekuefler we released J2CL couple of months back.
Today @sarsonj asked about Guava Eventbus (https://github.com/google/j2cl/issues/29) and I pointed to your project.
It might be a good time to port the generator to APT.
